src/compiler/scala/tools/nsc/transform/async/StateSet.scala

```diff
@@ -0,0 +1,44 @@
+/*
+ * Scala (https://www.scala-lang.org)
+ *
+ * Copyright EPFL and Lightbend, Inc.
+ *
+ * Licensed under Apache License 2.0
+ * (http://www.apache.org/licenses/LICENSE-2.0).
+ *
+ * See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.
+ */
+
+package scala.tools.nsc.transform.async
+
+import java.util
+import java.util.function.{Consumer, IntConsumer}
+
+import scala.collection.JavaConverters.{asScalaIteratorConverter, iterableAsScalaIterableConverter}
+
+// Set for StateIds, which are either small positive integers or -symbolID.
+final class StateSet {
+  private val bitSet = new java.util.BitSet()
+  private val caseSet = new util.HashSet[Integer]()
```

Comment:
How common are states above 1024? Would it make sense to allocate `caseSet` lazily on demand?

Reply:
Nice idea, done.